controllers/flowcollector_controller.go

```diff
@@ -83,5 +101,6 @@ func (r *FlowCollectorReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&appsv1.Deployment{}).
 		Owns(&appsv1.DaemonSet{}).
 		Owns(&corev1.Service{}).
+		Owns(&osv1alpha1.ConsolePlugin{}).
```

Comment:
What happens when it doesn't run in the context of openshift? (e.g. on kind) Creating the ConsolePlugin resource will fail obviously, but without breaking the rest?

Reply:
Thanks, I did not think about regular kubernetes cases.

I was thinking about adding a flag to enable/disable the plugin deployment, what do you think ?
Do you think we should try to automatically detect that the pod is running under OpenShift ?

Reply:
actually, I think the console itself can be deployed in non-openshift kube. So the best solution would be to detect if the ConsolePlugin crd is installed
(we can leave it aside at the moment if you prefer - just create a follow-up task then - but we must make sure it won't make the whole operator crashing ; it must still deploy / reconcile goflow and the like)

Reply:
I added detection to the controller, the controller looks for console group.

Note that this check is only done at startup, should be watch for created groups in case the console is deployed after the operator ?

Reply:
I'd say it's ok as it is, the console being created after the NOO would really be a corner case IMO.